homeassistant/components/tag/__init__.py

```diff
@@ -106,6 +107,15 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     return True
 
 
+class TagProtocol(Protocol):
```

Comment:
MQTT Tag seems to allow passing a `None` value as device_id. I believe that should not possible, am I right?

Reply:
Based on this code, I would say that neither tag_id nor device_id are compulsory.
The type hints were added in #52540

https://github.com/home-assistant/core/blob/f34de5072a53fe9b0d36dfe75fa697f4aa263376/homeassistant/components/tag/trigger.py#L34-L39